packages/block-library/src/image/image.js

```diff
@@ -142,6 +144,16 @@ export default function Image( {
 			.then( ( blob ) => setExternalBlob( blob ) );
 	}, [ id, url, isSelected, externalBlob ] );
 
+	// Focus the caption after inserting an image from the placeholder. This is
+	// done to preserve the behaviour of focussing the first tabbable element
+	// when a block is mounted. Previously, the image block would remount when
+	// the placeholder is removed. Maybe this behaviour could be removed.
+	useEffect( () => {
+		if ( url && ! prevUrl && isSelected ) {
+			captionRef.current.focus();
+		}
+	}, [ url, prevUrl ] );
```

Comment:
I understand now, thanks for the separate PR. I think this implementation is better but I wonder if focusing the block (the container) is actually better after selecting an image.

Reply:
Maybe @jasmussen has ideas.

Reply:
Can you walk me through the challenge?

This is what I see in the shipping product:

![before](https://user-images.githubusercontent.com/1204802/93761592-53916280-fc0e-11ea-8b5d-0ea0faae280d.gif)

This is what I see in this branch:

![after](https://user-images.githubusercontent.com/1204802/93761604-5724e980-fc0e-11ea-89db-996043fbad22.gif)

They seem the same, and seem okay to me. There's a piece I'm missing.

Reply:
@youknowriad To be honest, I would love if we focus the container instead. This makes much more sense to be and it's something I've proposed before as a general behaviour (focus container, and if it's editable it automatically sets the caret).
@jasmussen No behaviour change to master in the branch. The question is if the placeholder => image focus change to caption is good.

Reply:
Are we ok with merging this PR in the meantime?

Reply:
The focus-the-container vs. focus-the-caption-field sounds like a good accessibility question. Pinging @afercia to check this out.

Reply:
Thanks for the ping. To my understanding there's no practical change to the keyboard interaction, right? @ellatrix 

On a side note: I think we discussed but never established a good principle for the "initial focus" on the blocks. Generally, setting focus on a place that makes great part of the block UI "skipped" isn't ideal. Not to mention that The image block isn't fully operable with keyboard yet. For exampe, I can't find a way to move focus to the caption toolbar.